dashboard-server/queries/responses/response.test.js

```diff
@@ -57,34 +74,351 @@ describe('Response queries', () => {
+            survey.survey_name,
+            encodedToken,
+          );
+
+          const goodSurvey = survey;
+          const badSurvey = survey2;
+
+          const answers = responses.items[0].answers;
+
+          answers.forEach(el => el.surveyid.should.equal(goodSurvey.id));
+          answers.forEach(el => el.surveyid.should.not.equal(badSurvey.id));
+          answers.forEach(el =>
+            el.response.should.not.equal('Do not return me!'),
+          );
+        });
+      });
+
+      describe('pageSize', () => {
+        it('should return the specified maximum number of responses', async () => {
+          let pageSize = 2;
+
+          let maxResponses = await Response.all(
+            email,
+            surveyName,
+            after,
+            pageSize,
+          );
+
+          let answers = maxResponses.items[0].answers;
+          answers.length.should.equal(2);
+
+          pageSize = 1;
+
+          maxResponses = await Response.all(email, surveyName, after, pageSize);
+
+          answers = maxResponses.items[0].answers;
+          answers.length.should.equal(1);
+        });
+      });
+
+      describe('after', () => {
+        it('should return all new responses after a given token', async () => {
+          const after = token.getToken(timestamps[2], '126', 'ref');
+
+          const responsesAfterToken = await Response.all(
+            email,
+            surveyName,
+            after,
+          );
+
+          const answers = responsesAfterToken.items[0].answers;
+          answers.length.should.equal(3);
+        });
+
+        it('should return no new responses when on the last token', async () => {
+          const after = token.getToken(timestamps[3], '126', 'ref');
+
+          const responsesAfterToken = await Response.all(
+            email,
+            surveyName,
+            after,
+          );
+
+          responsesAfterToken.items[0].should.eql({
+            token: 'MjAyMi0wNi0wNiAxMDowMjowMCswMDowMA==',
+            answers: [],
+          });
+          responsesAfterToken.items[0].answers.length.should.equal(0);
+        });
+      });
+
+      describe('ROUTE /all', () => {
+        it('responds with a list of responses after a given token', async () => {
+          // first request
+          const firstResponse = await request(app)
+            .get(`/all?survey=${surveyName}&pageSize=25`) // no token needed here
+            .set('Accept', 'application/json')
+            .expect('Content-Type', /json/)
+            .expect(200);
+
+          firstResponse.statusCode.should.equal(200);
+          firstResponse.headers['content-type'].should.equal(
+            'application/json; charset=utf-8',
+          );
+
+          const answers = firstResponse.body.items[0].answers;
+          answers.length.should.equal(4);
+
+          let tokenReceived = firstResponse.body.items[0].token;
+          tokenReceived.should.equal('MTk3MC0wMS0wMSAwMDowMDowMCswMDowMA==');
```

Comment:
I wouldn't even assert what the token looks like. You just need to assert that the token _works_ - that it gets you a new set of data. Does that make sense?